pkg/operator/resource/resourceapply/generic.go

```diff
@@ -37,17 +41,22 @@ type ApplyResult struct {
 	Error   error
 }
 
-// TODO provide ways to provide cached getters
 type ClientHolder struct {
```

Comment:
nit: would be nice to have something like `NewClientHolder()` so one don't have to do `(&resourceapply.ClientHolder{}).WithKubernetes(kubeClient),`

Reply:
or just `NewKubeClientHolder()` ?